### ISSUE TITLE
Enhance visual features/functionality of raylib map

### ISSUE THREAD

**Issue by seanrryan** [uchicago-cs/chiventure#714]
~~This task will hopefully bring us nearer to finalizing the configuration and setup of raylib. This task will require some changes to CMake files and possibly other parts of the chiventure. I definitely need to do some more research to figure out what components of chiventure need to be modified to reflect these changes we have in mind.~~

This opening description is from my initial task (modifying chiventure's CMake files to make raylib a formal dependency) which we determined not to follow through with.

**Comment by seanrryan** [comment]
After discussing with the rest of the team, we determined that working specifically with the CMake configuration would not be productive at this stage in the quarter. 

For the remainder of the sprint my responsibility has shifted instead to building off of #727 to further enhance visual features/functionality of the map.

**Comment by seanrryan** [comment]
The map example pull request has been merged—I hope to make some improvements to this example that could translate to greater ease of use going forward. A few ideas:

- Support for keyboard- or mouse-controlled cursor: this could be used to exit the map screen or bring up important information about rooms. For example, with a cursor, a possible feature to add in the future could highlight the selected room and display its name (or perhaps "???" if it has not been explored yet).

- Scrolling and magnification: For large and/or irregular room layouts it may be difficult to present the entire map on the screen. Support for moving the map around the screen, or "zoom in" and "zoom out" features could allow for the player to manipulate the screen to see rooms.

These are just a few ideas I have been thinking about for improving upon our existing map implementation.

**Comment by seanrryan** [comment]
I chose to work on a keyboard-controlled cursor to add to the existing map example, which I saw to be a fairly straightforward enhancement that would integrate well with the existing work so far. The keyboard-based cursor is useful not only for allowing cursor control without a mouse, but it also stays true to original text-based adventure games. The way raylib handles shapes in the environment is very useful for this feature going forward, because the coordinates of the circle/cursor are stored in the map environment. For example, a function could be written that could detect whether the cursor overlaps with one of the rooms or the player, which could then produce a new object to display information about what is being selected.
